drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2023 Jan Henke <Jan.Henke@taujhe.de>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sparkfun_serlcd
+
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/auxdisplay.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(auxdisplay_serlcd, CONFIG_AUXDISPLAY_LOG_LEVEL);
+
+/*
+ * | in ASCII, used to begin a display command
+ */
+#define SERLCD_BEGIN_COMMAND 0x7C
+
+/*
+ * special command for the underlying display controller
+ */
+#define SERLCD_BEGIN_SPECIAL_COMMAND 0xFE
+
+union auxdisplay_display_state {
+	uint8_t byte;
+	struct __packed {
+#ifdef CONFIG_LITTLE_ENDIAN
+		uint8_t padding: 5;
+		bool power: 1;
+		bool cursor: 1;
+		bool blinking: 1;
+#else
+		bool blinking: 1;
+		bool cursor: 1;
+		bool power: 1;
+		uint8_t padding: 5;
+#endif
```

Comment:
you sure this is a good idea? normally I'd expect the driver file to `#define` the bit offset, keep whatever data you need in data and then prepare the command to send via i2c based on that. You could just do that, move these fields in `auxdisplay_serlcd_data`, saves you from having to manually deal with packed structures, unions etc...

Reply:
just to clarify where I stand on this: fine by me if @nordicjm is happy with it, but I don't think this is good practice and should be changed to work like most other drivers with defines for bit fields etc

Reply:
The define and bit offset suggestion sounds like a good improvement

Reply:
The whole things is the result of previous review comments. As far as I am concerned, I am for the most simple solution.

Reply:
Do you have any driver which does the define/offset thing as a blueprint?

Reply:
I think I now have a non-packed version, albeit without offset but more explicit bit setting

Reply:
Hey, yeah good stuff, that's what I meant, just use some defines for the bit definitoins now, ideally matching the name in the datasheet or manual or something.